contents/blog/missing-recordings.md

```diff
@@ -0,0 +1,56 @@
+
+We decided to use the second metric because it would let us strive to 0% missing recordings. If you're curious, you can read the conversation that led to this decision in [this GitHub issue](https://github.com/PostHog/posthog/issues/5478).
+
+In the first run of the metric, it showed that we we're missing ~15% of recordings across our cloud users.
+
+## Checking Sentry for clues
+
+Now that we had a metric, we started hunting for clues. What was causing the missing recordings?
+
+The first place we looked was Sentry. There were a couple of errors in the ingestion pipeline that seemed suspicious. The most notable was a "Max data upload size exceeded" error that was firing a few thousand times a day. This seemed like it could be an obvious cause of missing recordings. If some data was not being uploaded, because it was too big, it would make sense that we could not get a 'full snapshot'.
+
+To figure out what was causing this error, we looked at data that had been uploaded that was just below our size threshold of 20mb. It looked like large data urls of images were the cause.
+
+To test this out, we made a filter that would remove data URLs from any payload that exceeded a 5mb threshold, and it would replace the image with a simple SVG.
+
+After deploying [the fix](https://github.com/PostHog/posthog-js/pull/317), the Sentry error went away, but the missing recording only went down a few percent.
+
+## Looking deeper
+
+What was causing the rest of the missing recordings? With no more clues, it came time to dig through the mechanics of the code.
+
+When a recording started, we took a 'full snapshot' of the page, and from that point forward, we would send up only the data that changed in the recording. For some reason, the missing recordings would often have the later data, but no full snapshot.
+
+Why were the full snapshots missing? Presumably, if the later data was being sent up, the snapshot code should've already run. To further solidify this line of thinking, the determining the changes on the page fully depended on the snapshot existing in memory, so it must've already taken a full snapshot.
+
+After a day or so of pounding our heads on the keyboard, we realized that recordings were being split when a user was inactive for 30 minutes. But when we split the recording, we were not sending up a new full snapshot.
+
+We made [a fix](https://github.com/PostHog/posthog-js/pull/318) for this, and the missing recordings started to quickly drop.
+
+## Slow moving metrics
+
+Because this fix went into posthog-js, and different customers upgrade at different speeds, it took a while to see the full effect of the fix in our metrics. However, after about a month of waiting, the 15% missing recordings metric dropped to <1%.
```

Comment:
I feel like we're missing a conclusion here:

1) What did we learn from fixing this problem?
2) What feedback, if any, did we receieve from customers?

Reply:
100% agree. Just added a conclusion that touched on the other quality improvements that we made, and the impact we've seen on usage.